### DISCUSSION TITLE
[React] useDismiss via keyboard in Popover doesn't return focus to trigger element

### DISCUSSION THREAD

**Discussion by EliteUser** [floating-ui/floating-ui#2493] [Q&A]
Hello again!

Implementing popover component and faced following issue with keyboard dismiss behavior.
I created a [reproduction](https://codesandbox.io/s/clever-brown-cxffws?file=/src/App.tsx)
My component accepts trigger element via `children` property. Inside component I take children and clone them with passing reference props via `getReferenceProps`.
So when my trigger element is a `button`, everything works fine - I open popover with keyboard, then press `esc` and focus returns to trigger element. 
But when I wrap `button` in some `div` (or anything else) - focus doesn't return to button, because, as I understand, focus is trying to return to div.

Is there any way to handle this case inside component? Or it is a bug?

**Comment by atomiks** [comment]
The reference ref needs to be attached to the button yes
